### PR TITLE
feat: debug pkg

### PR DESCRIPTION
**Pull request by Pantani**

### CODE REVIEW

File: examples/flags/main.go [hunk 1]
```diff
@@ -4,10 +4,10 @@ import (
 	"context"
 	"fmt"
 
+	"flags/cmd"
+
```
Comment (jeronimoalbi) [inline examples/flags/main.go:8]: I believe this imports should still be placed at the bottom of the imports

File: test/debug/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# debug
+
+This package is a helper for you can debug your app outside the Ignite using devel or another debuger.
```
Comment (jeronimoalbi) [inline test/debug/README.md:3]: If it makes sense maybe something like:

```suggestion
This package is a helper that you could use to debug your Ignite App by running it as a standalone binary independently of Ignite CLI.
```

**Comment by jeronimoalbi** [comment]
@Pantani it might be a good idea to add a "Caveat" or similar section to the readme explaining that this method doesn't support debugging interactions with the CLI. If I got it correctly this method allows debugging running App commands independently of the CLI, which means that Ignite doesn't know the PID of the App, so it won't be able to attach to it and because of that debugged Apps don't be able to communicate with Ignite to for example use the Client API.

At some point I tried to create a debug app to run a debugger on apps that are connected with Ignite but I had blocking issues because the plugins system and the debugger use stdout. But maybe we could implement an "intermediate" solution using the approach in this PR and supporting dynamic attach in Ignite CLI.

By the way, it might be good to scaffold debug support by default.

**Comment by Pantani** [comment]
> @Pantani it might be a good idea to add a "Caveat" or similar section to the readme explaining that this method doesn't support debugging interactions with the CLI. If I got it correctly this method allows debugging running App commands independently of the CLI, which means that Ignite doesn't know the PID of the App, so it won't be able to attach to it and because of that debugged Apps don't be able to communicate with Ignite to for example use the Client API.
> 
> At some point I tried to create a debug app to run a debugger on apps that are connected with Ignite but I had blocking issues because the plugins system and the debugger use stdout. But maybe we could implement an "intermediate" solution using the approach in this PR and supporting dynamic attach in Ignite CLI.
> 
> By the way, it might be good to scaffold debug support by default.

nice, added

**Review by jeronimoalbi — APPROVED** [review]

**Pull request merged**